pkg/store/store.go

```diff
@@ -443,36 +442,43 @@ func (store *OpenTSDBStore) checkMetricNames(metricNames []string, fullBlock boo
 	return allowed, warnings, nil
 }
 
-func convertOpenTSDBResultsToSeriesResponse(respI *opentsdb.QueryRespItem) (*storepb.SeriesResponse, error) {
-	samples := make([]prompb.Sample, 0)
-	for _, dp := range respI.GetDataPoints() {
-		samples = append(samples, prompb.Sample{Timestamp: dp.Timestamp, Value: dp.Value.(float64)})
-	}
+func convertOpenTSDBResultsToSeriesResponse(respI *opentsdb.QueryRespItem) (*storepb.SeriesResponse, int, error) {
 	seriesLabels := make([]storepb.Label, len(respI.Tags))
 	i := 0
 	for k, v := range respI.Tags {
 		seriesLabels[i] = storepb.Label{Name: k, Value: v}
 		i++
 	}
 	seriesLabels = append(seriesLabels, storepb.Label{Name: "__name__", Value: respI.Metric})
-	enc, cb, err := encodeChunk(samples)
-	if err != nil {
-		return nil, err
-	}
-	var minTime, maxTime int64
-	if len(samples) != 0 {
-		minTime = samples[0].Timestamp
-		maxTime = samples[len(samples)-1].Timestamp
+
+	// Turn datapoints into chunks (Prometheus's tsdb encoding)
+	dps := respI.GetDataPoints()
+	chunks := []storepb.AggrChunk{}
+	for i := 0; i < len(dps); {
+		c := chunkenc.NewXORChunk()
+		a, err := c.Appender()
+		if err != nil {
+			return nil, 0, err
+		}
+		var minTime int64
+		// Maximum 120 datapoints in a chunk
+		for ; i < len(dps) && (minTime == 0 || i % 120 != 0); i++ {
```

Comment:
Can the comment say why 120?

Reply:
Added that it's a thanos limit for now and a link to the slack thread directly.